src/Component/RuleOverview/RuleOverview.tsx

```diff
@@ -0,0 +1,167 @@
+import { Symbolizers, SymbolizersProps } from '../Symbolizers/Symbolizers';
+import CardViewUtil from '../../Util/CardViewUtil';
+import { FilterOverview } from '../FilterOverview/FilterOverview';
+
+// i18n
+export interface RuleOverviewLocale {
+  ruleTitle: string;
+}
+
+// default props
+interface RuleOverviewDefaultProps {
+  /** Locale object containing translated text snippets */
+  locale: RuleOverviewLocale;
+  /** The callback when the style changed. */
+  onRuleChange: (rule: GsRule) => void;
+  /** The callback when a view change (request) was triggered. */
+  onChangeView: (view: string, indices: number[]) => void;
+}
+
+// non default props
+export interface RuleOverviewProps extends Partial<RuleOverviewDefaultProps> {
+  /** Reference to internal data object (holding schema and example features). */
+  data?: Data;
+  /** A GeoStyler-Style object. */
+  rule: GsRule;
+  /** The passthrough props for the RuleFieldContainer component. */
+  ruleFieldContainerProps?: Partial<RuleFieldContainerProps>;
+  /** The passthrough props for the Symbolizers component. */
+  symbolizersProps?: Partial<SymbolizersProps>;
+}
+
+export const RuleOverview: React.FC<RuleOverviewProps> = ({
+  rule,
+  data,
+  onRuleChange = () => {},
+  onChangeView = () => {},
+  ruleFieldContainerProps,
+  symbolizersProps,
+  locale = en_US.GsRuleOverview,
+}) => {
+
+  const [stateRule, setStateRule] = useState<GsRule>(rule);
+
+  const onNameChange = (name: string) => {
+    const newRule: GsRule = {...stateRule, name};
+    setStateRule(newRule);
+    onRuleChange(newRule);
+  };
+
+  const onMinScaleChange = (minScale: number) => {
+    let newRule: GsRule = {...stateRule};
+    if (!newRule.scaleDenominator) {
+      newRule.scaleDenominator = {};
+    }
+    newRule.scaleDenominator.min = minScale;
+    setStateRule(newRule);
+    onRuleChange(newRule);
+  };
+
+  const onMaxScaleChange = (maxScale: number) => {
+    let newRule: GsRule = {...stateRule};
+    if (!newRule.scaleDenominator) {
+      newRule.scaleDenominator = {};
+    }
+    newRule.scaleDenominator.max = maxScale;
+    setStateRule(newRule);
+    onRuleChange(newRule);
+  };
+
+  const onSymbolizersChange = (symbolizers: GsSymbolizer[]) => {
+    let newRule: GsRule = {...stateRule, symbolizers};
+    setStateRule(newRule);
+    onRuleChange(newRule);
+  };
+
+  const onEditSymbolizerClick = (symbolizerId: number) => {
+    onChangeView(CardViewUtil.SYMBOLIZERVIEW, [symbolizerId]);
+  };
+
+  const onEditFilterClick = () => {
+    onChangeView(CardViewUtil.FILTEREDITVIEW, []);
+  };
+
+  const ruleFieldContainerPropsOverwrites = {
+    rendererProps: {
+      symbolizers: rule.symbolizers,
+      data: data
+    }
+  };
+  const mergedRuleFieldContainerProps = _merge(ruleFieldContainerProps, ruleFieldContainerPropsOverwrites);
+
+  const symbolizersPropsOverwrites = {
+  };
+  const mergedSymbolizersProps = _merge(symbolizersProps, symbolizersPropsOverwrites);
```

Comment:
Can be replaced with spread operator syntax.

Reply:
solved in https://github.com/geostyler/geostyler/pull/1507/commits/3e057aa39664e4e89fcdd875c32c996f8ed757fd